test/tests/functional/pbs_reservations.py

```diff
@@ -1123,3 +1123,128 @@ def test_reservations_with_expired_subjobs(self):
         # Check status of the parent-job using qstat -fx once reservation ends
         self.server.expect(JOB, {'job_state': 'F', 'substate': '91'},
                            extend='x', id=jid)
+
+    def test_ASAP_resv_with_multinode_job_array(self):
+        """
+        Test multinode job array with ASAP reservation
```

Comment:
Can you please explain why do you need multi-node jobs? This test isn't testing anything related to multiple nodes. Consider making it multi-vnode test instead.
Also, in this test you are submitting two different array jobs, creating 2 ASAP reservations out of it. Can you please write a docstring about what you are trying to test with multiple array jobs and reservations that can not be tested with one.